cache/util/fsutil_test.go

```diff
@@ -0,0 +1,34 @@
+package util
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tonistiigi/fsutil"
+)
+
+// TestSetErrorPath ensures that modifying the os.PathError from fsutil.Stat
+// modifies the error message appropriately.
+//
+// This method of modifying the path error is an implementation
+// detail of how the error is returned. It isn't necessarily the
+// case that this will always work if the underlying library changes.
+func TestSetErrorPath(t *testing.T) {
+	// Temporary directory to reduce the chance of using stat on a real file.
+	dir, err := os.MkdirTemp("", t.Name())
+	require.NoError(t, err)
+	t.Cleanup(func() { os.RemoveAll(dir) })
+
+	// Random path that shouldn't exist.
+	fpath := path.Join(dir, "a/b/c")
+	_, err = fsutil.Stat(fpath)
+	require.Error(t, err)
+
+	require.Contains(t, err.Error(), "a/b/c")
```

Comment:
`.ErrorContains` like in your other update. Surprised that testifylint does not catch this.